### PR TITLE
Do not offer to add default switch case when it handles `null` and an underlying value

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/50983

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs [hunk 1]
```diff
@@ -103,4 +102,36 @@ public static bool IsDefault(IPatternOperation pattern)
...
+    public static bool HasBothNullAndUnderlyingValueCases(ISwitchExpressionOperation operation)
+    {
+        var type = operation.Value.Type;
+        var underlyingType = type.IsNullable(out var underlying) ? underlying : type;
+
+        var hasNullArm = false;
+        var hasUnderlyingTypeArm = false;
+
+        foreach (var arm in operation.Arms)
+        {
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs:115]: There might be an UnwrapNullable helper for this

File: src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs [hunk 2]
```diff
@@ -103,4 +102,36 @@ public static bool IsDefault(IPatternOperation pattern)
...
+                ITypePatternOperation typePattern => typePattern.MatchedType,
+                IDeclarationPatternOperation declarationPattern => declarationPattern.MatchedType,
+                _ => null
+            };
+
+            if (SymbolEqualityComparer.Default.Equals(matchedType, underlyingType))
+                hasUnderlyingTypeArm = true;
+        }
+
+        return hasNullArm && hasUnderlyingTypeArm;
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs:135]: Put in for loop so it can return immediately?

**Review by CyrusNajmabadi — APPROVED** [review]
Minor suggestions. Lgtm!

File: src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs (hunk 2 above)
Comment (DoctorKrolic) [reply src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs:135]: Didn't understand this one. We can only be sure we have both arms/cases only if we loop through all of them and check each. How can that be short-circuited?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs]: I pushed a change where i moved this into the loop. ;)

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**